test/stream_data_test.exs

```diff
@@ -527,6 +527,28 @@ defmodule StreamDataTest do
     end)
   end
 
+  property "optional_map/2" do
+    data_with_map = optional_map(%{integer: integer(), binary: binary()}, [:integer])
+    data_with_keyword = optional_map([integer: integer(), binary: binary()], [:integer])
+
+    Enum.each([data_with_map, data_with_keyword], fn data ->
+      check all map <- data do
+        assert map_size(map) in [1, 2]
+
+        assert map
+               |> Map.keys()
+               |> MapSet.new()
+               |> MapSet.subset?(MapSet.new([:integer, :binary]))
+
+        if Map.has_key?(map, :integer) do
+          assert is_integer(Map.fetch!(map, :integer))
+        end
```

Comment:
This is  tricky test. It will pass if the generated map never has the `:integer` key.

I think a better test here would be to take a significant amount of values from the generator (with `Enum.take/2` if you want), like 100 values, and check that at least one generated map has the `:integer` key. It's statistically almost sure that it will 🙃

Reply:
👍🏻 makes perfect sense, will do!

Reply:
Done :)